src/main/java/hudson/tasks/junit/TestResultAction.java

```diff
@@ -179,8 +179,8 @@ public synchronized TestResult getResult() {
             skipCount = r.getSkipCount();
         }
         long d = System.nanoTime() - started;
-        if (d > 500000000L && LOGGER.isLoggable(Level.WARNING)) {
```

Comment:
500000000 is 5e8, or 0.5 seconds. I adjusted the code to use `TimeUnit` to make it clearer, but I could switch it to 500 * 1000 * 1000 or whatever you think is clearest.

Reply:
(I also wonder if this threshold is too low for `WARNING` and it should be increased to 1-2 seconds, maybe logging at `FINE` for stuff between 0.5-1 seconds)

Reply:
I like the idea of setting the threshold higher for logging at less than 2 seconds, but approve this pull request with or without that change.